### PR TITLE
feat/add leave trip button

### LINKED ISSUE #298: add the possibility to leave a trip
- as a participant, I want to be able to leave a trip
- TBD: reassign the creator if the creator leaves the trip

### PR DESCRIPTION
**Pull request by mirceagavrizi**
## Summary

This PR introduces the possibility to leave a trip. An user needs to click on the 3 points in the top right of the trip card and the dropdown now contains the "Leave" button.

## Changes

- **Screens/UI:** Added the Leave button and removed the spacer in the add trip screen
- **Bug Fixes/Improvements:** Addresses #298 

## Checklist

Ensure all / most of these are checked before the pull request is submitted.
- [x] This PR resolves #298 .
- [x] Tests have been added for new functionality.
- [x] Screenshots or UI changes have been attached.
- [x] Documentation has been updated.

##  Testing

- **Manual Testing:**
    - [x] Tested on emulator / physical device.
- **Unit Tests**
    - [x] Added units tests for this

##  Related Issues/Tickets

Closes #298 

## Screenshots 

https://github.com/user-attachments/assets/33521a88-75b3-438c-bec5-e91865777a6a

## 💡 Additional Context

The placement of the button could be discussed at the next meeting.

### CODE REVIEW

File: app/src/main/java/com/android/voyageur/ui/overview/Overview.kt [hunk 1]
```diff
@@ -386,6 +394,45 @@ fun TripItem(
...
+                        Toast.makeText(
+                                context,
+                                context.getString(R.string.trip_left_text),
+                                Toast.LENGTH_SHORT)
+                            .show()
+                      },
+                      onFailure = { error ->
+                        Toast.makeText(
+                                context,
+                                "Failed to leave the trip: ${error.message}",
```
Comment (apana03) [inline app/src/main/java/com/android/voyageur/ui/overview/Overview.kt:422]: This hardcoded string should also be extracted into the strings.xml file.

**Review by apana03 — COMMENTED** [review]
Good job for the very clear implementation! Only had one minor comment regarding a hardcoded string. Also I can't help but notice that the line coverage is only at 60% percent, you should try to increase it. Once that is done I have no problem merging this PR.

File: app/src/main/java/com/android/voyageur/ui/overview/Overview.kt (hunk 1 above)
Comment (mirceagavrizi) [reply app/src/main/java/com/android/voyageur/ui/overview/Overview.kt]: You're right, changed it.

**Comment by mirceagavrizi** [comment]
Regarding the line coverage, I tried for a while to test that specific part, but I couldn't mock the behavior. The overall coverage for this file will still be above 80% and I will raise another PR next week for adding overall tests in which I will also include this one.

**Pull request merged**